### PR TITLE
refactor(example): enhance example for public inputs validation

### LINKED ISSUE #1132: Add examples

### PR DESCRIPTION
**Pull request by IAvecilla**
> [!CAUTION]
> Content of thi PR is OK but we had to make another branch.
> [the new PR is this one](https://github.com/yetanotherco/aligned_layer/pull/1245)

Update the public input validation example to support both `SP1` and `Risc0` implementations. Additionally, improve the steps and documentation to make it easier to follow and execute.

## How to test

You can test it locally following the commands in `examples/validating-public-input/README.md` or use holesky following the official documentation in `docs/guides/3_validating_public_input.md`.

### CODE REVIEW

File: docs/3_guides/3_validating_public_input.md [hunk 1]
```diff
@@ -2,178 +2,145 @@
 
 In some applications, it is crucial to ensure that a third party has performed a computation correctly and to make use of the result of that computation. To achieve this, the third party must first interact with Aligned and obtain the `AlignedVerificationData`, a receipt indicating that the proof of the computation was verified correctly. The application should then receive both the `AlignedVerificationData` and the result of the computation. After confirming that the proof was verified by Aligned, it must check that the posted result matches the one committed in the `AlignedVerificationData`.
```
Comment (uri-99) [inline docs/3_guides/3_validating_public_input.md]: `should first interact with Aligned and obtain the AlignedVerificationData` , is this the SDK ?

File: docs/3_guides/3_validating_public_input.md [hunk 2]
```diff
@@ -2,178 +2,145 @@
...
 
-This guide demonstrates how to validate a Risc0 proof using the Aligned SDK. The Risc0 program in this example is a Fibonacci sequence calculator. It generates a public input that corresponds to the last two Fibonacci numbers of the sequence, taken modulo 7919. Our goal is to validate, within a smart contract, that the public input commitments match these two numbers.
+This guide demonstrates how to validate a Risc0 and SP1 proofs using the Aligned SDK. The program in this example is a Fibonacci sequence calculator. It generates a public input that corresponds to the number of fibonacci being calculated and the last two Fibonacci numbers of the sequence, taken modulo 7919. Our goal is to validate, within a smart contract, that the public input commitments match these two numbers.
```
Comment (uri-99) [inline docs/3_guides/3_validating_public_input.md]:
```suggestion
This guide demonstrates how to validate Risc0 and SP1 proofs using the Aligned SDK. The program in this example is a Fibonacci sequence calculator. It generates a public input that corresponds to the number of fibonacci being calculated and the last two Fibonacci numbers of the sequence, taken modulo 7919. Our goal is to validate, within a smart contract, that the public input commitments match these two numbers.
```

File: docs/3_guides/3_validating_public_input.md [hunk 3]
```diff
@@ -2,178 +2,145 @@
...
 
-You can find examples on how to generate proofs in the [generating proofs guide](4_generating_proofs.md).
+To generate the proof required for this example, run the following commands:
 
-To generate the proof needed to try this example, run `make generate_risc_zero_fibonacci_proof`.
+- For **Risc0**: `make generate_risc0_fibonacci_proof`
+- For **SP1**: `make generate_sp1_fibonacci_proof`
 
-Once finished, you will see the program id, the two last fibonacci numbers of the sequence and the result of the verification like so:
+Once completed, you will see output that includes the program ID, the public inputs (which are the initial number and the last two Fibonacci numbers of the sequence), and the verification result, like so:
```
Comment (uri-99) [inline docs/3_guides/3_validating_public_input.md]:
```suggestion
Once completed, you will see output that includes the program ID, the public inputs (which are the number of steps in the sequence and the last two Fibonacci numbers of the sequence), and the verification result, like so:
```

File: examples/validating-public-input/Makefile [hunk 4]
```diff
@@ -1,11 +1,44 @@
...
+verify_sp1_batch_inclusion:
+	@. ./contracts/.env  && . ./contracts/validate_batch_inclusion.sh $(FIBONACCI_VALIDATOR_ADDRESS) $(DATA_FILE_NAME) SP1
+
+verify_risc0_batch_inclusion:
+	@. ./contracts/.env  && . ./contracts/validate_batch_inclusion.sh $(FIBONACCI_VALIDATOR_ADDRESS) $(DATA_FILE_NAME) Risc0
+
+verify_risc0_local_batch_inclusion:
+	@. ./contracts/.env.devnet  && . ./contracts/validate_batch_inclusion.sh $(FIBONACCI_VALIDATOR_ADDRESS) $(DATA_FILE_NAME) Risc0
+
+verify_sp1_local_batch_inclusion:
```
Comment (uri-99) [inline examples/validating-public-input/Makefile]:
```suggestion
anvil_verify_risc0_batch_inclusion:
	@. ./contracts/.env.devnet  && . ./contracts/validate_batch_inclusion.sh $(FIBONACCI_VALIDATOR_ADDRESS) $(DATA_FILE_NAME) Risc0

anvil_verify_sp1_batch_inclusion:
```
or
```suggestion
verify_risc0_batch_inclusion_devnet:
	@. ./contracts/.env.devnet  && . ./contracts/validate_batch_inclusion.sh $(FIBONACCI_VALIDATOR_ADDRESS) $(DATA_FILE_NAME) Risc0

verify_sp1_batch_inclusion_devnet:
```

File: examples/validating-public-input/Makefile [hunk 5]
```diff
@@ -1,11 +1,44 @@
...
+	@. ./contracts/.env.devnet  && . ./contracts/validate_batch_inclusion.sh $(FIBONACCI_VALIDATOR_ADDRESS) $(DATA_FILE_NAME) SP1
 
 deploy_fibonacci_validator:
 	@. ./contracts/.env && . ./contracts/deploy.sh
+
+deploy_local_fibonacci_validator:
```
Comment (uri-99) [inline examples/validating-public-input/Makefile]: same here

File: examples/validating-public-input/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+This example is designed to do either with SP1 or risc0 proofs these are the commands to use depending on which verifier want to be used.
+
+### Risc0
+
+1. `make generate_risc0_fibonacci_proof`
+
+2. `make submit_fibonacci_risc0_proof_devnet`
+
+> The command will log the file where all the aligned verification data was saved, save the name since it will be necessary.
+3. `deploy_local_fibonacci_validator`
```
Comment (uri-99) [inline examples/validating-public-input/README.md]:
```suggestion
3. `make deploy_local_fibonacci_validator`
```

File: examples/validating-public-input/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+> The command will log the address where the validator was deployed, save it for the next command.
+4. `make verify_risc0_local_batch_inclusion <FIBONACCI_VALIDATOR_ADDRESS> <DATA_FILE_NAME>`
```
Comment (uri-99) [inline examples/validating-public-input/README.md]: i think make targets dont work like this, did you mean the .sh ? or adding the `VAR=` ?

File: examples/validating-public-input/contracts/validate_batch_inclusion.sh [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+fi
+
+proof_commitment=$(jq -r '.proof_commitment' "../aligned-integration/batch_inclusion_data/$DATA_FILE_NAME.json")
+pub_input_commitment=$(jq -r '.pub_input_commitment' "../aligned-integration/batch_inclusion_data/$DATA_FILE_NAME.json")
+program_id_commitment=$(jq -r '.program_id_commitment' "../aligned-integration/batch_inclusion_data/$DATA_FILE_NAME.json")
+proof_generator_addr=$(jq -r '.proof_generator_addr' "../aligned-integration/batch_inclusion_data/$DATA_FILE_NAME.json")
+batch_merkle_root=$(jq -r '.batch_merkle_root' "../aligned-integration/batch_inclusion_data/$DATA_FILE_NAME.json")
+merkle_proof=$(jq -r '.merkle_proof' "../aligned-integration/batch_inclusion_data/$DATA_FILE_NAME.json")
+verification_data_batch_index=$(jq -r '.verification_data_batch_index' "../aligned-integration/batch_inclusion_data/$DATA_FILE_NAME.json")
+pub_input=$(jq -r '.pub_input' "../aligned-integration/batch_inclusion_data/$DATA_FILE_NAME.json")
```
Comment (uri-99) [inline examples/validating-public-input/contracts/validate_batch_inclusion.sh]: I would remove the `.json` to let the user put the full file name

File: examples/validating-public-input/README.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+Where `FIBONACCI_VALIDATOR_ADDRESS` is the address of the deployed validator contract and `DATA_FILE_NAME` the name of the file where the aligned verification data was saved (without the extension `.json`).
+
```
Comment (uri-99) [inline examples/validating-public-input/README.md:52]: Missing what the appropriate response for the previous command looks like

**Review by uri-99 — CHANGES_REQUESTED** [review]
Also, should this PR be pointed at `testnet`?

File: docs/3_guides/3_validating_public_input.md (hunk 2 above)
Comment (IAvecilla) [reply docs/3_guides/3_validating_public_input.md]: Fixed here: [20b8aae](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/20b8aae8b3512f3225b119c142cfb7840e284a1f)!

File: docs/3_guides/3_validating_public_input.md (hunk 3 above)
Comment (IAvecilla) [reply docs/3_guides/3_validating_public_input.md]: Fixed here: [20b8aae](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/20b8aae8b3512f3225b119c142cfb7840e284a1f)!

File: examples/validating-public-input/Makefile (hunk 4 above)
Comment (IAvecilla) [reply examples/validating-public-input/Makefile]: Fixed here: [20b8aae](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/20b8aae8b3512f3225b119c142cfb7840e284a1f)!

File: examples/validating-public-input/Makefile (hunk 5 above)
Comment (IAvecilla) [reply examples/validating-public-input/Makefile]: Fixed here: [20b8aae](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/20b8aae8b3512f3225b119c142cfb7840e284a1f)!

File: examples/validating-public-input/README.md (hunk 6 above)
Comment (IAvecilla) [reply examples/validating-public-input/README.md]: Fixed here: [20b8aae](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/20b8aae8b3512f3225b119c142cfb7840e284a1f)!

File: examples/validating-public-input/README.md (hunk 7 above)
Comment (IAvecilla) [reply examples/validating-public-input/README.md]: Yeah, it was a mistake I meant to do the `VAR=` notation. Fixed here: [20b8aae](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/20b8aae8b3512f3225b119c142cfb7840e284a1f)!

File: examples/validating-public-input/contracts/validate_batch_inclusion.sh (hunk 8 above)
Comment (IAvecilla) [reply examples/validating-public-input/contracts/validate_batch_inclusion.sh]: Fixed here: [20b8aae](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/20b8aae8b3512f3225b119c142cfb7840e284a1f)!

File: examples/validating-public-input/README.md (hunk 9 above)
Comment (IAvecilla) [reply examples/validating-public-input/README.md:52]: I added the output for each command to be easier to follow, it's added here [d46cc30](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/d46cc3023886b614612faac4613bc73ecb27ee50)

File: docs/3_guides/3_validating_public_input.md (hunk 1 above)
Comment (IAvecilla) [reply docs/3_guides/3_validating_public_input.md]: Added here [d46cc30](https://github.com/yetanotherco/aligned_layer/pull/1195/commits/d46cc3023886b614612faac4613bc73ecb27ee50)!

**Review by uri-99 — CHANGES_REQUESTED** [review]
Single proofs emitted will not get included in a batch, so in devnet it is necesarry to send more proofs in parallel. We should add this detail in the docs since this PR is going to testnet.
After this it is an approve

**Review by PatStiles — APPROVED** [review]

**Comment by uri-99** [comment]
Closing this PR because it is being solved in this other rebased [PR](https://github.com/yetanotherco/aligned_layer/pull/1245)

**Pull request closed without merging**